microprofile/WFLY-19835_microprofile_reactive_messaging_otel_with_kafka.adoc

```diff
@@ -0,0 +1,176 @@
+:connector: Kafka
+:connector-lower: kafka
+:issue-id: WFLY-19835
+
+= Integrate OpenTelemetry with the MP Reactive Messaging {connector} connector
+:author:            Kabir Khan
+:email:             your.email@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+
+
+== Overview
+
+SmallRye Reactive Messaging's {connector} connector allows for integration with OpenTelemetry tracing so that we can track messages sent to and received from {connector} via this connector.
+
+In SmallRye Reactive Messaging, a set of MicroProfile Config properties are used to control whether tracing happens on a reactive messaging channel. These are:
+
+* `mp.messaging.connector.smallrye-{connector-lower}.tracing-enabled` - This is a connector level property. At present this is set to `false` in a `ConfigSource` set up by the integration with an `ordinal` value set to the maximum value. The result of this is that the user can never enable tracing to happen (since any attempt to set this property to `true` would be overridden by the mentioned `ConfigSource`).
+* The above property may in turn be overridden by the following channel level properties.
+** `mp.messaging.outgoing.<channel-name>.tracing-enabled` - Channel level property, controlling the OpenTelemetry tracing integration for an outgoing channel.
+** `mp.messaging.incoming.<channel-name>.tracing-enabled` - Channel level property, controlling the OpenTelenetry integration for an incoming channel.
+
+The way the SmallRye Reactive Messaging configuration system works is that it will first check for relevant channel level properties. If those are not present, it will look for the related connector level property, and if that is not available it will use a default.
+
+Internally in SmallRye Reactive Messaging, the default value for the tracing integration is that it is always enabled for a channel if not specified with above properties.
+
+This RFE is about allowing users to enable tracing via OpenTelemetry in WildFly, and also to be able to continue to ban it.
+
+
+=== User Stories
+
+Users wish to be able to trace messages sent to/from {connector} via reactive messaging using MicroProfile Telemetry.
+
+== Issue Metadata
+
+* https://issues.redhat.com/browse/{issue-id}[{issue-id}] - Integrate OpenTelemetry with the MP Reactive Messaging {connector} connector
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-19846[WFLY-19846] - Promote MicroProfile Telemetry 2.0 to WildFly Standard
+
+
+=== Affected Projects or Components
+* https://github.com/wildfly/wildfly/ - All integration work is done here
+* https://github.com/smallrye/smallrye-reactive-messaging - contains the implementation of the Reactive Messaging connectors. No work is expected to be needed here, unless bugs are found.
+
+=== Other Interested Projects
+None to my knowledge
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
```

Comment:
Why not managed domain?

EAP XP doesn't support domain mode, but that is not due to any fundamental technical limitations. And this is WildFly functionality / a WildFly proposal and AFAIK there's no reason this wouldn't work. The relevant extensions are all supported in WildFly domain mode.

https://github.com/wildfly/wildfly/blob/main/galleon-pack/galleon-feature-pack/wildfly-feature-pack-build.xml#L93-L107